.ci.yaml

```
@@ -194,6 +194,13 @@ targets:
       lint_android: "false"
       lint_host: "true"
     timeout: 60
+    runIf:
+      - DEPS
+      - .ci.yaml
+      - tools/**
+      - ci/**
+      - "**.c"
+      - "**.cc"
```

I think this also needs to run on changes to header files right? Here and below.

```suggestion
      - "**.cc"
      - "**.h"
```

---

You're right.  Should I filter out `third_party` from these checks?

---

Hm, you explicitly added a test for `third_party` so I guess not #26722

---

Yeah, the script already skips source files in third_party.

Morning brain is also remembering that we generate and analyze source for flat buffers and shader bindings, so this should also run on changes to `*.fbs`, `*.frag`, and `*.vert`.

---

Sorry for my shader ignorance, so changes to those files generate .cc files that are being linted?

The list of files it's linting:
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8799745439669169009/+/u/test:_lint_ios_debug_sim/stdout

https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8799745440095255937/+/u/test:_lint_host_debug/stdout

---

I think the generated source might be in header files, which is why they might not show up in the list of .cc files being analyzed. It's why we have to do builds before linting: https://flutter.googlesource.com/recipes/+/refs/heads/main/recipes/engine/engine_lint.py#91.

---

Updated.  That's a lot of files, but at least it will be skipped on dart and java -only commits 🙂

---

It _is_ being skipped on some PRs, so that's something.
https://github.com/flutter/engine/pull/37187
https://github.com/flutter/engine/pull/37212